### ISSUE TITLE
Enhancement to Job based analysis run 

### ISSUE THREAD

**Issue by satyakiran-opsmx** [argoproj/argo-rollouts#2482]
# Summary

To provide rich UI facility for the Job based Analysis Run. Following are the enhancements for the Rollout:

1. Ability to accept additional data in the Analysis Run resource when the metric provider is configured as a Job.
2. Ability to give extension support for the Analysis Run Widget to render the analysis run result.  

# Use Cases

1. When the provider is configured as a Job and wants to show the analysis run result in a report:

    - Either the provider can send the externalURL, if the report is hosted on a different namespace. This URL will be loaded in a new tab when the user clicks on Analysis Run Widget in Rollout Dashboard UI.
    - Or the provider can develop an extension and configure it in the Rollout, so that, on click of the Analysis Run Widget, 
the extension will be loaded with the additional data passed as part of the run result.

In short, a provider can load the analysis run report page either by sending an external URL to show from a different application or by developing an extension.

Explained in more detail in the below document.

https://docs.google.com/document/d/1p0srOZR48njjdrHdmChCs6i6Lh-7NlKouM43_nIRbtE/edit#heading=h.va4qgkpyy1cv

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritize the issues with the most 👍.

**Comment by balajisiva** [comment]
Since the community is moving from metrics provider-based analysis to job-based analysis, this is a mandatory requirement to make the analysis accessible for the developer/user to quickly decide to roll-forward or back.

**Comment by zachaller** [comment]
I think a good majority of these improvements do make sense, I want to spend some time as well thinking about the message passing back to the analysis controller and would love for this proposal to also be brought up during a contributors meeting so we can start a conversation a bit about it how to implement that. From the contributors meeting we can also continue the disscussion in https://cloud-native.slack.com/archives/C044E9ZFS4V

**Comment by shraddhaopsmx** [comment]
Hi @zachaller, we have very recently updated the google document to explain a little more on what we are proposing on the UI end under the UI proposal heading, can you please give the doc a relook and see if the enhancements(the UI ones) would fit well here?

**Comment by zachaller** [comment]
> Hi @zachaller, we have very recently updated the google document to explain a little more on what we are proposing on the UI end under the UI proposal heading, can you please give the doc a relook and see if the enhancements(the UI ones) would fit well here?

I like the idea of an external metric link, however I think I would prefer it to be done slightly different, I would like to maybe handel it like argocd does and use a little external link icon within the analysis button when `externalURL` is present. That way we have the option of having some high level stuff still within the rollouts UI but also allowing an external link click out. I also think letting the user know it's an external link via the icon is nice addition.

**Comment by satyakiran-opsmx** [comment]
We like your idea @zachaller. we will go with the external link icon ![image](https://user-images.githubusercontent.com/73812688/219662788-0d68779a-a046-406c-88d9-63cb24f595b9.png) within the Analysis button in this fashion:
![image](https://user-images.githubusercontent.com/73812688/219663900-9b97fe44-e0a6-43a0-9ae9-560c9818ea77.png) 
(image from Argo CD App Details Tree)

Regarding the UI plugin - we are planning for that. Our main idea is to support any metric provider who doesn't have report page available via `externalURL`. This way the UI plugin can help render the analysis report in Rollout Dashboard without any external server dependency. Please let us know your thoughts on this approach.

**Comment by zachaller** [comment]
> Regarding the UI plugin - we are planning for that. Our main idea is to support any metric provider who doesn't have report page available via `externalURL`. This way the UI plugin can help render the analysis report in Rollout Dashboard without any external server dependency. Please let us know your thoughts on this approach.

Yes I think that makes sense with maintaining the current support for viewing the state of analysis runs in the UI with as much detail as is feasible and the if the provider gives an `externalURL` back as the metadata it would show the external link to provide a more custom experiance.

**Comment by zachaller** [comment]
Closing in favor of plug-ins

**Comment by shaoye** [comment]
do we have a link to the plug-in?

**Comment by zachaller** [comment]
Not sure if you mean specifically the ops mx plugin or the plugin feature https://github.com/argoproj-labs/rollouts-opsmx-metric-plugin/tree/feature/add-plugin-functionality and https://argo-rollouts.readthedocs.io/en/release-1.5/plugins/
